### ISSUE TITLE
Dealing with inappropriate parameter input values

### ISSUE THREAD

**Issue by codykallen** [open-source-economics/PFLCM#11]
As should be obvious, the various policy parameters and take-up assumptions should have limitations on the values they can take. For example, the `include_*` parameters should only be able to take values of 0 or 1, and the maximum duration should not be set to zero or negative values. 

These restrictions/limitations generally take two forms: 
1) values that break the code
 - Ex: setting `maxduration_days = -10` would cause the code to fail, as no unit can take negative durations
 - Ex: setting `waiting_period = 500` causes the code to fail, as no unit can take annual leave exceeding some a year

2) values that don't break the code but cause incorrect results
 - Ex: setting `maxduration_days = 500` would not break the code, but the limit would be nonbinding for all units and beyond modeling capabilities
 - Ex: setting `waiting_period = 30` would not break the code, but the modeling for the "reduced employer push" would not apply correctly
 - Ex: setting `include_military = 0.5` would not break the code, but the parameter is only intended to modeling the inclusion or exclusion of this type of leave, so intermediate values don't make sense

I am planning to add some modifications to the beginning of the `run_allassumptions.do` file to ensure that inappropriate values are not passed. I will certainly add assert statements that stop the program if code-breaking values are passed, but there is an open question on how to deal with inappropriate parameter values. I am inclined to simply use a warning that the value exceeds the recommended bounds. 

@begitis @MattHJensen

**Comment by codykallen** [comment]
After testing, the following input values cause the code to break:
`maxduration_days < 21`
`waiting_period > 33`

These occur because the code calculates take-up, duration and other information based on subsets of the FMLA data. If the maximum duration is set too low or the waiting period set too high, these subsets may be empty, in which case the `collapse` code fails. I see two potential options to correct this:
1. Add assert statements that prevent the code from running if `maxduration_days` is set too low or if `waiting_period` is set too high. 
2. Move the `collapse` statements in the second major section of `run_anyassumptions.do` into `if` statements; if they fail, then the desired estimates would be replaced by zeros. 

In addition, the following parameter values would not break the code but would produce nonsensical results:
`maxduration_days > 242`
`waiting_period > 20`
`include_* != 0 and include_* != 1`
`overall_takeup > 1`
`overall_takeup < 0`
`reduce_* < 0`
`reduce_* > 1`
`frac_employerpush > 1`
`frac_employerpush < 0`

For these inappropriate inputs, I plan to add warning statements at the beginning of the `run_anyassumptions.do` file, but the code will still run.

**Comment by codykallen** [comment]
In the previous comment, I made a suggestion regarding providing zeros if there are no observations in the FMLA data meeting the conditions. The code to implement this would be:

```
count if CONDITIONS
if r(N)>0 {
collapse VAROFINTEREST if CONDITIONS
rename VAROFINTEREST VAROFINTEREST2
}
else {
gen VAROFINTEREST2 = 0
collapse VAROFINTEREST2
}
```

**Comment by codykallen** [comment]
This has been resolved by #18.
